sstables_loader.cc

```diff
@@ -218,11 +240,31 @@ future<> sstable_streamer::stream(std::function<void(unsigned)> on_streamed) {
     co_await stream_sstables(full_partition_range, std::move(_sstables), std::move(on_streamed));
 }
 
+bool tablet_sstable_streamer::tablet_in_scope(locator::tablet_id tid) const {
+    if (_stream_scope == stream_scope::all) {
+        return true;
+    }
+
+    const auto& topo = _erm->get_topology();
+    for (const auto& r : _tablet_map.get_tablet_info(tid).replicas) {
+        if (_stream_scope == stream_scope::node && topo.is_me(r.host)) {
+            return true;
+        }
```

Comment:
why continue in the loop body here?
Should this be structured as follows :
```c++
        switch (_stream_scope) {
        case stream_scope::node
            if (topo.is_me(r.host)) {
                return true;
            }
            break;
        case stream_scope::rack:
            if (topo.get_location(r.host) == topo.get_location()) {
                return true;
            }
            break;
        case stream_scope::dc:
            if (topo.get_datacenter(r.host) == topo.get_datacenter()) {
                return true;
            }
            break;
        }
    }
    return false;
```

Reply:
It's longer and (I think) compiler is good enough at optimizing such dead branches out.
Other than that, _stream_scope is enum and not adding `case stream_scope::all:` will be noticed by compiler. And since this case was explicitly handled up above, what should be done here?

Reply:
>And since this case was explicitly handled up above, what should be done here?

break?

Reply:
But it would be incorrect

Reply:
> But it would be incorrect

Break from the switch statement, not from the for loop.

> It's longer and (I think) compiler is good enough at optimizing such dead branches out.
Other than that, _stream_scope is enum and not adding case stream_scope::all: will be noticed by compiler. And since this case was explicitly handled up above, what should be done here?

I'm fine with an if-else-if ladder on the _stream_scope conditions, and the lower conditions can be checked in the nested blocks, like:
```c++
    if (_stream_scope == stream_scope::node) {
        if (topo.is_me(r.host)) {
            return true;
        }
    } else if (_stream_scope == stream_scope::rack) {
        if (topo.get_location(r.host) == topo.get_location()) {
            return true;
        }
    } else if etc.
}
```

Reply:
I would say that if-ladder the way it's here results in shorter code with less conditional jumps: https://godbolt.org/z/1rP3nWbdY

Reply:
The nested if-ladder is even 2 bytes shorter :)
https://godbolt.org/z/bx7cEzP6z